### PR TITLE
remove old API deprecated since CGAL 4.4

### PR DESCRIPTION
**Pull request by sloriot**
The old API is dangerous as seen in #1292 
It requires the approval of @gdamiand 

TODO:
- [x] get the approval of Guillaume
- [x] announce the breaking change in the changelog of CGAL (either in 4.9 or 4.10, depending on the date we get the approval)

### CODE REVIEW

**Comment by lrineau** [comment]
[TODO: ][*] modify the changelog, to announce the removal of the deprecated API.

[*]: https://github.com/CGAL/cgal/pull/1293#issue-166780579

**Comment by gdamiand** [comment]
I approve (I didn't check if all the deprecated code are removed).

**Comment by lrineau** [comment]
Then the remaining TODO is the announcement in the changelog of 4.9.

File: Installation/changes.html [hunk 1]
```diff
@@ -172,6 +172,9 @@ <h2 id="release4.9">Release 4.9 </h2>
     <code>insert_cell_2_in_cell_3()</code>, <code>insert_dangling_cell_1_in_cell_2()</code>, <code>is_insertable_cell_1_in_cell_2()</code>,
     <code>is_insertable_cell_2_in_cell_3()</code>, <code>is_removable()</code>, <code>remove_cell()</code>)
     which are now member functions in the <code>CombinatorialMap</code> concept.</li>
+    <li>It is not longer possible to use the old API switched on by defining the macro
+    <code>CGAL_CMAP_DEPRECATED</code>. This API was deprecated since CGAL 4.4
```
Comment (lrineau) [inline Installation/changes.html]: Typo: Missing `.` after `CGAL 4.4`.

**Comment by lrineau** [comment]
Error in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-106.shtml#Polyhedron_Demo
> ../Polyhedron_Demo/Plugins/Operations_on_polyhedra/Scene_combinatorial_map_item.cpp:363:34: error: **no member named 'is_free'** in 'CGAL::Dart<3, CGAL::Combinatorial_map_storage_1<3, CGAL::internal_IOP::Item_with_points_and_volume_info<CGAL::Mesh_3::Robust_intersection_traits_3<CGAL::Epick>, CGAL::Polyhedron_3<CGAL::Mesh_3::Robust_intersection_traits_3<CGAL::Epick>, Polyhedron_demo_items<int>, CGAL::HalfedgeDS_default, std::allocator<int> > >, std::allocator<int> > >'

**Comment by sloriot** [comment]
The error was in #1292

**Pull request merged**